src/Tribe/Integrations/Freemius.php

```diff
@@ -0,0 +1,307 @@
+
+	/**
+	 * Stores the ID for the Freemius application.
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $freemius_id = '3841';
+
+	/**
+	 * Stores the slug for the Freemius application.
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $slug = 'event-tickets';
+
+	/**
+	 * Stores the name for the Freemius application.
+	 *
+	 * @since  TBD
+	 *
+	 * @var string
+	 */
+	private $name = 'Event Tickets';
+
+	/**
+	 * Store the value from the 'page' in the request.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private $page;
+
+	/**
+	 * Store the value from the 'tab' in the request.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private $tab;
+
+	/**
+	 * Tribe__Tickets__Integrations__Freemius constructor.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$this->setup();
+	}
+
+	/**
+	 * Performs setup for the Freemius integration singleton.
+	 *
+	 * @since  TBD
+	 */
+	public function setup() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		global $pagenow;
+
+		$this->page = tribe_get_request_var( 'page' );
+
+		$valid_page = [
+			Tribe__Settings::$parent_slug,
+			Tribe__App_Shop::MENU_SLUG,
+			'tribe-help',
+		];
+
+		if ( class_exists( 'Tribe__Events__Aggregator__Page' ) ) {
+			$valid_page[] = Tribe__Events__Aggregator__Page::$slug;
+		}
+
+		if ( 'plugins.php' !== $pagenow && ! in_array( $this->page, $valid_page, true ) ) {
+			return;
+		}
+
+		// If the common that loaded doesn't include Freemius, let's bail.
+		if ( ! tribe()->offsetExists( 'freemius' ) ) {
+			return;
+		}
+
+		$should_load = true;
+
+		// Check if Freemius integration is disabled.
+		if (
+			( defined( 'TRIBE_NO_FREEMIUS' ) && true === TRIBE_NO_FREEMIUS )
+			|| true === (bool) getenv( 'TRIBE_NO_FREEMIUS' )
+		) {
+			$should_load = false;
+		}
+
+		/**
+		 * Allows third-party disabling of the integration.
+		 *
+		 * @since  TBD
+		 *
+		 * @param bool $should_load Whether the Freemius integration should load.
+		 */
+		$should_load = apply_filters( 'tribe_tickets_integrations_should_load_freemius', $should_load );
+
+		if ( ! $should_load ) {
+			return;
+		}
+
+		/** @var Tribe__Freemius $freemius */
+		$freemius = tribe( 'freemius' );
+
+		$this->instance = $freemius->initialize( $this->slug, $this->freemius_id, $this->public_key, [
+			'menu'           => [
+				'slug'    => $this->page,
+				'account' => true,
+				'support' => false,
+			],
+			'is_premium'     => false,
+			'has_addons'     => false,
+			'has_paid_plans' => false,
+		] );
+
+		$this->instance->add_filter( 'connect_url', [ $this, 'redirect_settings_url' ] );
+		$this->instance->add_filter( 'after_skip_url', [ $this, 'redirect_settings_url' ] );
+		$this->instance->add_filter( 'after_connect_url', [ $this, 'redirect_settings_url' ] );
+		$this->instance->add_filter( 'after_pending_connect_url', [ $this, 'redirect_settings_url' ] );
+
+		tribe_asset( $this->object_class, 'tribe-tickets-freemius', 'freemius.css', [], 'admin_enqueue_scripts' );
```

Comment:
I've found a hacky way around the plugin icon issue (and it relates the the freemius.css enqueue. Rather than the above line, we can do:

```php
		add_filter( 'fs_plugin_icon_event-tickets', function( $icon ) {
			add_action( 'admin_footer', function() {
				echo '<link rel="stylesheet" href="' . Tribe__Tickets__Main::instance()->plugin_url . '/src/resources/css/freemius.css"/>';
			});

			return $icon;
		} );
```

When the plugin icon template is being output, it is specific to the plugin being loaded in the moment. We can then include a stylesheet in the admin_footer that provides the CSS file with the correct icon. A similar change would be needed in TEC like so:

```php
		add_filter( 'fs_plugin_icon_the-events-calendar', function( $icon ) {
			add_action( 'admin_footer', function() {
				echo '<link rel="stylesheet" href="' . Tribe__Events__Main::instance()->plugin_url . '/src/resources/css/freemius.css"/>';
			});

			return $icon;
		} );
```

Reply:
Now, if the freemius.css file needs to be included all the time, we can break the icon-specific rule into its own CSS file. Then we'd have something like:

```php
tribe_asset( $this->object_class, 'tribe-tickets-freemius', 'freemius.css', [], 'admin_enqueue_scripts' );

add_filter( 'fs_plugin_icon_event-tickets', function( $icon ) {
	add_action( 'admin_footer', function() {
		echo '<link rel="stylesheet" href="' . Tribe__Tickets__Main::instance()->plugin_url . '/src/resources/css/freemius-site-icon.css"/>';
	});

	return $icon;
} );
```

Reply:
```php
$this->instance->add_filter( 'plugin_icon', static function( $icon ) use ( $object_class ) {
	wp_enqueue_style( 'tribe-tickets-freemius', Tribe__Tickets__Main::instance()->plugin_url . '/src/resources/css/freemius.css' );

	return $icon;
}, 11, 6 );
```